lib/rule-tester/rule-tester.js

```diff
@@ -1236,35 +1236,36 @@ class RuleTester {
 							);
 						}
 
+						const actualLocation = {};
+						const expectedLocation = {};
+
 						if (hasOwnProperty(error, "line")) {
-							assert.strictEqual(
-								message.line,
-								error.line,
-								`Error line should be ${error.line}`,
-							);
+							actualLocation.line = message.line;
+							expectedLocation.line = error.line;
 						}
 
 						if (hasOwnProperty(error, "column")) {
-							assert.strictEqual(
-								message.column,
-								error.column,
-								`Error column should be ${error.column}`,
-							);
+							actualLocation.column = message.column;
+							expectedLocation.column = error.column;
 						}
 
 						if (hasOwnProperty(error, "endLine")) {
-							assert.strictEqual(
-								message.endLine,
-								error.endLine,
-								`Error endLine should be ${error.endLine}`,
-							);
+							actualLocation.endLine = message.endLine;
+							expectedLocation.endLine = error.endLine;
 						}
 
 						if (hasOwnProperty(error, "endColumn")) {
-							assert.strictEqual(
-								message.endColumn,
-								error.endColumn,
-								`Error endColumn should be ${error.endColumn}`,
+							actualLocation.endColumn = message.endColumn;
+							expectedLocation.endColumn = error.endColumn;
+						}
+
+						if (Object.keys(expectedLocation).length > 0) {
+							assert.deepStrictEqual(
+								actualLocation,
+								expectedLocation,
+								`Error location should be ${util.inspect(
+									expectedLocation,
+								)}, found ${util.inspect(actualLocation)}`,
```

Comment:
I would prefer if we will show the full actual location here for two reasons:

- If there is an error it helps with detecting the reported range
- It nudges developers to verify all location properties instead the subset they might be using.

```suggestion
								)}, found ${util.inspect(fullActualLocation)}`,
```